server/src/app.ts

```diff
@@ -22,83 +24,170 @@ app.get("/alive", (req: Request, res: Response) => {
+          steps,
+        }: {
+          title: string;
+          description: string;
+          categories: string[];
+          ingredients: string[];
+          steps: string[];
+        } = req.body;
+
+        const newItem: Item = {
+          title,
+          description,
+          categories,
+          ingredients,
+          steps,
+          createdAt: Math.floor(Date.now() / 1000),
+        };
+        const result: InsertOneResult<any> = await db
+          .collection("recipes")
+          .insertOne(newItem);
+        res
+          .status(200)
+          .json({ message: "recipe was added", postId: result.insertedId });
+      } catch (error) {
+        console.error("Error creating item:", error);
+        res.status(500).json({ error: "Failed to create item" });
+      }
+    });
+
+    //GET
+
+    app.get("/api/v1/items", async (req: Request, res: Response) => {
+      try {
+        const collection: Collection = db.collection("recipes");
+        const documents = await collection.find().toArray();
+
+        const recipes: Item[] = documents.map((recipe) => ({
+          title: recipe.title,
+          description: recipe.description,
+          categories: recipe.categories,
+          ingredients: recipe.ingredients,
+          steps: recipe.steps,
+          createdAt: recipe.createdAt,
+        }));
+
+        res.status(200).json(recipes);
+      } catch (error) {
+        console.error("Error getting items:", error);
+        res.status(500).json({ error: "Failed to get items" });
+      }
+    });
+
+    app.get("/api/v1/items/:id", async (req: Request, res: Response) => {
+      if (ObjectId.isValid(req.params.id)) {
+        try {
+          const collection: Collection = db.collection("recipes");
+          const document = await collection.findOne({
+            _id: new ObjectId(req.params.id),
+          });
+          res.status(200).json(document);
+        } catch (error) {
+          console.error("Error getting item:", error);
+          res.status(500).json({ error: "Failed to get items" });
+        }
+      } else {
+        res.status(500).json({ error: "Not valid document ID" });
+      }
+    });
+
+    //DELETE
+
+    app.delete("/api/v1/items/:id", async (req: Request, res: Response) => {
+      if (ObjectId.isValid(req.params.id)) {
+        try {
+          const collection: Collection = db.collection("recipes");
+          const document = await collection.deleteOne({
+            _id: new ObjectId(req.params.id),
+          });
+          res.status(200).json(document);
+        } catch (error) {
+          console.error("Error deleting item:", error);
+          res.status(500).json({ error: "Failed to delete item" });
+        }
+      } else {
+        res.status(500).json({ error: "Not valid document ID" });
+      }
+    });
+
+    // PATCH
+
+    app.patch("/api/v1/items/:id", async (req: Request, res: Response) => {
+      if (ObjectId.isValid(req.params.id)) {
```

Comment:
```
if (!ObjectId.isValid(req.params.id)) {
  return res.status(400).json({ error: "Not valid document ID" });
}

try {
  // ...
} catch (error) {
  // ...
}
```
keep the happy flow on top so that the code is more readable